src/Client/ClientBase.cpp

```diff
@@ -352,6 +352,16 @@ ASTPtr ClientBase::parseQuery(const char *& pos, const char * end, bool allow_mu
         res = parseQueryAndMovePosition(*parser, pos, end, "", allow_multi_statements, max_length, settings.max_parser_depth);
     }
 
+    /// We need to change the auth_data before the query will be formatted
+    if (const auto * create_user_query = res->as<ASTCreateUserQuery>())
+    {
+        if (create_user_query->auth_data->getType() == AuthenticationType::NO_PASSWORD && create_user_query->temporary_password)
```

Comment:
It doesn't look correct. `NO_PASSWORD` means a user was created without specifying credentials or explicitly with `NOT IDENTIFIED` clause. And it definitely doesn't look like what you're checking here.

Reply:
`NO_PASSWORD` is a default value of the AuthenticationType enum, and after that, we check that temporary_password is filled, so if it is, then it is us who did not fill in the password type and we should assign it here.

Reply:
This looks like a hack. If we create a new AST-like version of `AuthenticationData` it will be easy to make the code clearer.